README.md

```diff
@@ -59,9 +56,65 @@ This entails that any blocking call occurring during the write operation would b
+    * [Publication mode](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/eprosimaExtensions.html#publishmodeqospolicy): `SYNCHRONOUS_PUBLISH_MODE`.
+
+1. Setting `RMW_FASTRTPS_USE_QOS_FROM_XML` effectively overrides whatever configuration was set with `RMW_FASTRTPS_PUBLICATION_MODE`, setting the publication mode to Fast DDS' default publication mode unless specified otherwise using an XML file.
+
+There are two ways of telling a ROS 2 application which XML to use:
+
+1. Placing your XML file in the running directory under the name `DEFAULT_FASTRTPS_PROFILES.xml`.
+1. Setting environment variable `FASTRTPS_DEFAULT_PROFILES_FILE` to contain the path to your XML file (relative to the working directory).
+
+#### Applying different profiles to different entities
+
+`rmw_fastrtps` allows for the configuration of different entities with different QoS using the same XML file.
+For doing so, `rmw_fastrtps` locates profiles in the XML based on topic names abiding to the following rules:
+
+##### Creating publishers/subscriptions with different profiles
+
+To configure a publisher/subscription, define a `<publisher>`/`<subscriber>` profile with attribute `profile_name=topic_name`.
+If such profile is not defined, `rmw_fastrtps` attempts to load the `<publisher>`/`<subscriber>` profile with attribute `is_default_profile="true"`.
+
+##### Creating services with different profiles
+
+ROS 2 services contain a subscription for receiving requests, and a publisher to reply to them.
+`rmw_fastrtps` allows for configuring each of these endpoints separately in the following manner:
+
+1. To configure the request subscription, define a `<subscriber>` profile with attribute `profile_name=topic_name`, where topic name is the name of the service after mangling.
+If such profile is not defined, `rmw_fastrtps` attempts to load a `<subscriber>` profile with attribute `profile_name="service"`.
+If neither of the previous profiles exist, `rmw_fastrtps` attempts to load the `<subscriber>` profile with attribute `is_default_profile="true"`.
+1. To configure the reply publisher, define a `<publisher>` profile with attribute `profile_name=topic_name`, where topic name is the name of the service after mangling.
+If such profile is not defined, `rmw_fastrtps` attempts to load a `<publisher>` profile with attribute `profile_name="service"`.
+If neither of the previous profiles exist, `rmw_fastrtps` attempts to load the `<publisher>` profile with attribute `is_default_profile="true"`.
+
+##### Creating clients with different profiles
+
+ROS 2 clients contain a publisher to send requests, and a subscription to receive the service's replies.
+`rmw_fastrtps` allows for configuring each of these endpoints separately in the following manner:
+
+1. To configure the requests publisher, define a `<publisher>` profile with attribute `profile_name=topic_name`, where topic name is the name of the service after mangling.
```

Comment:
it would be great to explain how the topic names after mangling look

Reply:
+1, but since some of the mangling is ROS 2 specific, let's not repeat existing documentation.